api/clients/v2/payload_disperser.go

```diff
@@ -0,0 +1,222 @@
+package clients
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Layr-Labs/eigenda/api/clients/codecs"
+	"github.com/Layr-Labs/eigenda/api/clients/v2/verification"
+	dispgrpc "github.com/Layr-Labs/eigenda/api/grpc/disperser/v2"
+	"github.com/Layr-Labs/eigenda/common/geth"
+	core "github.com/Layr-Labs/eigenda/core/v2"
+	"github.com/Layr-Labs/eigenda/encoding"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+)
+
+// PayloadDisperser provides the ability to disperse payloads to EigenDA
+//
+// This struct is goroutine safe.
+type PayloadDisperser struct {
+	logger          logging.Logger
+	config          *PayloadDisperserConfig
+	codec           codecs.BlobCodec
+	disperserClient DisperserClient
+	certVerifier    verification.ICertVerifier
+}
+
+// BuildPayloadDisperser builds a PayloadDisperser from config structs
+func BuildPayloadDisperser(
+	logger logging.Logger,
+	payloadDisperserConfig *PayloadDisperserConfig,
+	disperserClientConfig *DisperserClientConfig,
+	// signer to sign blob dispersal requests
+	signer core.BlobRequestSigner,
+	// prover is used to compute commitments to a new blob during the dispersal process
+	//
+	// IMPORTANT: it is permissible for the prover parameter to be nil, but operating with this configuration
+	// puts a trust assumption on the disperser. With a nil prover, the disperser is responsible for computing
+	// the commitments to a blob, and the PayloadDisperser doesn't have a mechanism to verify these commitments.
```

Comment:
hmm is this necessary? could we not instead just create a prover implementation that queries the disperser service instead, and force the prover to be non-nil?

Reply:
This would be possible, but for now this is just working with how the disperser client is implemented.

When I have a chance, I want to go back and make some improvements like this one